pasqal_cloud/batch.py

```diff
@@ -120,45 +120,42 @@ def _load_configuration(
             conf_class = EmuFreeConfig
         return conf_class.from_dict(configuration)
 
-    def add_job(
+    def add_jobs(
         self,
-        runs: int = 100,
-        variables: Optional[Dict[str, Any]] = None,
+        jobs: List[CreateJob],
         wait: bool = False,
-    ) -> Job:
-        """Add and send a new job for this batch.
+    ) -> "Batch":
+        """Add some jobs to batch for execution on PCS and returns the updated batch.
+
+        The batch should not be `complete` otherwise the API will return an error.
+        The new jobs are appended to the `ordered_jobs` list attribute.
 
         Args:
-            runs: number of times the job is run on the QPU.
-            variables (optional): values for variables if sequence is parametrized.
-            wait: Whether to wait for the job to be done.
+            jobs: List of jobs to be added to the batch.
+            wait: If True, blocks until all jobs in the batch are done.
 
-        Returns:
-            - Job: the created job.
         """
-        job_data: Dict[str, Any] = {"runs": runs, "batch_id": self.id}
-        if variables:
-            job_data["variables"] = variables
         try:
-            job_rsp = self._client._send_job(job_data)
+            batch_rsp = self._client._add_jobs(self.id, jobs)
         except HTTPError as e:
             raise JobCreationError(e) from e
-        job = Job(**job_rsp, _client=self._client)
-        self.ordered_jobs.append(job)
+        batch = Batch(**batch_rsp, _client=self._client)
         if wait:
-            while job.status in ["PENDING", "RUNNING"]:
+            while any(
+                [job.status in {"PENDING", "RUNNING"} for job in batch.ordered_jobs]
+            ):
                 time.sleep(RESULT_POLLING_INTERVAL)
                 try:
-                    job_rsp = self._client._get_job(job.id)
+                    batch_rsp = self._client._get_batch(self.id)
                 except HTTPError as e:
                     raise JobFetchingError(e) from e
-                job = Job(**job_rsp)
-        return job
+                batch = Batch(**batch_rsp, _client=self._client)
```

Comment:
@MohamedAminMallek @Augustinio I had to change the interface; now the function returns a new instance instead of modifying in place the batch object. 

Ideally I would prefer to update the object in place but the previous implementation which set `self=Batch(...)` doesn't actually update the current object. We could update all attributes one by one based on the http response but I haven't found a nice pattern just yet. I'm open to ideas

Reply:
I find it a little strange that a batch instance method doesn't update the state of the instance and return a new one. Perhaps we could have a dedicated private method that takes a dictionary (HTTP response) and updates the state of the batch.


Out of scope:
Why do we have a cancel_batch method in the SDK client and another one in the batch class?

Reply:
Yeah I dislike it as well. I'll see what I can do.

>Why do we have a cancel_batch method in the SDK client and another one in the batch class?
It's mostly for convenience for the user to make it possible to cancel a batch whether one has a Batch instance or simply the batch_id at hand

Reply:
I introduced a `refresh` method and used to it to update the object in place in the different methods, let me know if that looks good to you!